src/TestTree.ts

```diff
@@ -0,0 +1,235 @@
+import { Graph, json } from 'graphlib';
+import { TestInfo, TestSuiteInfo } from 'vscode-test-adapter-api';
+import { ParseOutput } from './utils/tests_parsers';
+
+export interface Test {
+  kind: 'test' | 'suite';
+  id: string;
+  label: string;
+  file: string;
+  belongsTo: TestSuite;
+  description?: string;
+  tooltip?: string;
+  line?: number;
+  skipped?: boolean;
+  debuggable?: boolean;
+  errored?: boolean;
+  message?: string;
+}
+
+export interface TestSuite {
+  kind: 'test' | 'suite';
+  id: string;
+  label: string;
+  belongsTo: TestSuite | undefined;
+  description?: string;
+  tooltip?: string;
+  file?: string;
+  line?: number;
+  debuggable?: boolean;
+  errored: boolean;
+  message?: string;
+}
+
+export interface ExportOptions {
+  scope: 'ALL' | 'NODE_ID';
+  nodeId?: string;
+}
+
+/*
+  Tree-based data structure which models all projects, directories and files as a graph.
+
+  Can translate the tree structure to a TestSuiteInfo (Test Explorer type), see export method.
+*/
+export class TestTree {
+  private graph: Graph;
+  private readonly root: TestSuite;
+  private readonly workspaceName;
+
+  constructor(workspaceName: string) {
+    this.graph = new Graph();
+    this.workspaceName = workspaceName;
+    this.root = {
+      id: 'ExUnit_suite_root',
+      belongsTo: undefined,
+      kind: 'suite',
+      label: `ExUnit ${this.workspaceName}`,
+      description: `Test suite for ${this.workspaceName}`,
+      tooltip: 'Test Suite',
+      errored: false,
+      file: undefined,
+    };
+    this.graph.setNode('ExUnit_suite_root', this.root);
+  }
+
+  public getNode(nodeId: string): Test | TestSuiteInfo {
+    return this.graph.node(nodeId);
+  }
+
+  public addTest(test: Test): Test {
+    const v = test.belongsTo.id || this.root.id;
```

Comment:
what does `v` and `w` mean?

Reply:
In graph theory, v and w are standard names for the tail and head of an edge.

https://github.com/dagrejs/graphlib/wiki/API-Reference#hasEdge

Reply:
Didn't know that, thanks :)